### PR TITLE
chore: get rid of type/pylint ignore statements

### PR DESCRIPTION
**Pull request by regisb**
1. Child model classes inherited the insert and update methods, but with a different signature. This violated the Liskov principle, which is explained here: https://stackoverflow.com/a/54359797/356528 To bypass this, we migrated most of the Content class to a separate base class, and insert/update methods are no longer inherited.

2. We removed type ignores in serializers, which were simply missing a generic type definition.

3. We also updated the APIClient, but we failed to properly type it, because we don't know how to use "_MonkeyPatchedWSGIResponse".

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -3,15 +3,14 @@
...
+from typing import Any
 
 import mongomock
 import pytest
-from django.http.response import HttpResponse
-from django.test import Client
 from pymongo import MongoClient
 
+from .client import APIClient
+
```
Comment (taimoor-ahmed-1) [inline tests/conftest.py:13]: should we be using relative imports in a small app like ours? there is another relative import in `tests/test_views/test_flags.py`.

Comment (regisb) [reply tests/conftest.py:13]: Oh sweet! I had not realized that we could use absolute imports in test_utils. I'll move the client module to test_utils and use an absolute import.

File: tests/test_views/test_flags.py [hunk 2]
```diff
@@ -2,12 +2,12 @@
 
 from unittest.mock import Mock, patch
 
-from django.test import Client
-
 from forum.models import Contents, Users
```
Comment (Faraz32123) [inline tests/test_views/test_flags.py:6]: we can remove the space, at line 6.

**Review by Faraz32123 — APPROVED** [review]
LGTM 👌

**Pull request merged**